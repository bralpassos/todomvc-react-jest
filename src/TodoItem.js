import React from 'react';

export default class TodoItem extends React.Component {

  renderItems = () => this.props.items.map((item, index) => (
      <li key={index}>
          <div className="view">
              <label>{item}</label>
          </div>
      </li>
    ));

    render() {
        return (
            <section className="main">
            <ul className="todo-list">
              {this.props.items.length === 0 ? '' : this.renderItems()}
            </ul>
          </section>
        );
    }
}