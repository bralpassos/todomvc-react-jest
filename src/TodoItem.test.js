import React from 'react';
import renderer from 'react-test-renderer';
import TodoItem from './TodoItem.js';

test('Não deve renderizar nada quando passa uma lista vazia', () => {
    const items = [];
    const component = renderer.create(<TodoItem items={items} />);
    expect(component).toMatchSnapshot();
});

test('Deve renderizar uma lista quando passa um item', () => {
    const items = ['Item'];
    const component = renderer.create(<TodoItem items={items} />);
    expect(component).toMatchSnapshot();
});

test('Deve renderizar uma lista com 2 ítens quando forem passados 2 ítens', () => {
    const items = ['Item1','Item2'];
    const component = renderer.create(<TodoItem items={items} />);
    expect(component).toMatchSnapshot();
});