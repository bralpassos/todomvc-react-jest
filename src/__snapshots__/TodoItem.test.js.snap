// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar uma lista com 2 ítens quando forem passados 2 ítens 1`] = `
<section
  className="main"
>
  <ul
    className="todo-list"
  >
    <li>
      <div
        className="view"
      >
        <label>
          Item1
        </label>
      </div>
    </li>
    <li>
      <div
        className="view"
      >
        <label>
          Item2
        </label>
      </div>
    </li>
  </ul>
</section>
`;

exports[`Deve renderizar uma lista quando passa um item 1`] = `
<section
  className="main"
>
  <ul
    className="todo-list"
  >
    <li>
      <div
        className="view"
      >
        <label>
          Item
        </label>
      </div>
    </li>
  </ul>
</section>
`;

exports[`Não deve renderizar nada quando passa uma lista vazia 1`] = `
<section
  className="main"
>
  <ul
    className="todo-list"
  >
    
  </ul>
</section>
`;
